### ISSUE TITLE
Updating a parent definition makes "branches"

### ISSUE THREAD

**Issue by Mycheze** [jzohrab/lute-v3#104]
**Description**

I don't know how else to describe the issue, but whenever you add a new word via a child, there's a duplicate definition. It's saved under the child AND the parent.

![image](https://github.com/jzohrab/lute-v3/assets/71797685/bc46baaa-57b2-4792-8326-88fdf419791f)
In the example image, the word šířit was added via the child šíří. The parent/child relationship is fine, but they BOTH have the definition saved.

This means that if I later decide to update šířit via another child (for example, if I come across a brand-new meaning in another context and want to add it to the parent), the original child will STILL have the old definition as well as the new one. This means that I'd need to edit the child to remove the definition.

I believe this is unintended behavior, which is why I chose to make this a bug report rather than feature request.

**To Reproduce**

Steps to reproduce the behavior, e.g.:

1. Click on brand-new child word that doesn't have a parent in the database yet
2. Add the parent
3. Add the definition
4. Have two definitions for 1 parent term

**Screenshots**

I've got tons of these built up.
![image](https://github.com/jzohrab/lute-v3/assets/71797685/8f545852-773e-40b5-bfaf-af24440cb4de)

**Extra software info, if not already included in the Description:**

- OS (e.g., iOS, windows): Pop_OS
- Browser (e.g., chrome, safari): Chromium
- How you've installed Lute (Docker, python, source): Python via pip
- Version: 3.0.6

**Comment by jzohrab** [comment]
This is actually intended behaviour :-) but I can see how it feels odd.

The reason for this: suppose I add a term and a parent at the same time, and enter a translation.  I save the term, and then immediately click it again.  If the description were only saved in the parent, then the term translation would be blank, which looks like some data has somehow been lost -- it would feel like a bug.

As I use Lute more, I don't think of the my definitions etc as a "dictionary", but rather as a scratchpad of notes.  I don't mind about the definition being dup'd, it just there to help me out.

> This means that if I later decide to update šířit via another child (for example, if I come across a brand-new meaning in another context and want to add it to the parent), the original child will STILL have the old definition as well as the new one. This means that I'd need to edit the child to remove the definition.

Yes, that's right -- I couldn't think of a way to make that better, and it felt like it could get messy to try to manage it differently.

**Comment by Mycheze** [comment]
I think it's be good to be able to view and edit parents in the same window then. I agree that it's nice to be able to add notes to children (to give specific usage notes or something), but I think that if you ever want for users to be able to share databases and entries, duplicated definitions, especially if one is outdated, is going to be weird and confusing.

**Comment by jzohrab** [comment]
Some misc notes, some copied from discord:

I’ve thought about this and I can’t see how to have the data go to the parent without it getting confusing, or messy to code/test.

* For example: I have a new term XX with new parent x. If I put in a translation for XX, and that was only saved with x, then when I reopen the form for Xx again the translation would be blank. That would feel like data loss.
* If I pull the translation for x into the form for XX, and I edit XX, should that change the parent?  If yes, then there’s no way to add a new XX-only translation, if no, we have the same problem.
* I wouldn’t want to add a “parent translation” text box to the form, it will confuse people, and gets hard to manage if parents are added and removed.  Old issue #15 (only set translation for parent) has some notes.
* If we only save the translation in the parent, then the term listing would likely need to pull in the parent translations, as noted in issue #15.  That's fine, that's just the query.  Would the CSV export also need to include that?
* How would we handle CSV imports, where the translation is set for the child term?  Would it only update the child, only update the parent (and potentially create inconsistencies there, if you have multiple children in the same file updating the parent)?

**Comment by cblanken** [comment]
I think a good first step for this issue would be to add a global (or per language) setting to disable the automatic duplication of the term definitions to the parent. Maybe even add a checkbox to the Term edit form. That way the user can configure the default behavior they prefer as well as use the checkbox to handle exception  for individual terms.

**Comment by jzohrab** [comment]
@cblanken - possibly yep, but I'm reluctant to get into it :-)  Every option means more user tweaking the settings, and another code path, though it's minor.  The questions I listed in my prior comment still apply.  Especially the first point, re something feeling like data loss.

I've never been bothered by the duplicated data, and it simplified me getting reading quickly, because I'd enter in data and it gets copied wherever it might be needed, even if not 100% accurate.  That was good enough for me ... but of course that's my own preference!

In summary, after thinking about it again, I still can't see how to make it dead simple to use.  To view and edit parents in the same window as Ben suggested above would make the form pretty busy.  I agree that the current situation is imperfect, but I feel that pushing this idea along wouldn't add much value to users, and could make things more complicated.

I wonder if there's a simpler solution that could help, e.g. something like a tool to clean up data:  "show children that have the same definition as their parents" and "clean these ones up", for example.  I'd be much more interested in this approach, again knowing it's not perfect, but at least it seems like it's clearly feasible, limited scope, easily understood by users, no interruption to existing flow, etc.

**Comment by Mycheze** [comment]
My primary problem is that I frequently need to edit a parent, usually because I improperly save something or forget to add a tag.

But if I've already closed the window, I now need to click into the child, delete it and save, then click into the child and then parent and THEN finally make the edit.

I'm not really sure why showing an editable parent box and child box would make things too cluttered. 

I'm attaching a potential layout which I think isn't too cluttered, clearer and much more useful.
![luteparentchildstuff](https://github.com/user-attachments/assets/9c3a9320-787d-449f-a7e8-cb7c18213067)

But a "delete identical children entries" does sound like a good first step. I would certainly use it.

**Comment by jzohrab** [comment]
Ah yeah that's a hassle, I've been fine with such cruft but can see how it would bother you.

I was thinking that the extra text boxes for parent terms would get cluttered, but that was in terms of the current layout.  Your alternative layout is nice but quite a departure from the current state.  Understood that this is just a sketch of possibilities.  Suffice it to say that it gets very fiddly on the code side!  Having dynamic forms is def stretching my abilities.

For "delete identical children descriptions", that _should_ be relatively quick to implement.  I opened #463 to get some initial ideas down.

**Comment by Mycheze** [comment]
Yep. And I'm happy to do more sketches if it's helpful. I'm no designer, but I enjoy it.

And I'm getting a little worried about the UX of Lute being difficult, even for people who have been using it for a long time (me).
